src/Agent.Worker/TaskDefinitionManager.cs

```diff
@@ -0,0 +1,291 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker.Handlers;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using System.Linq;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    [ServiceLocator(Default = typeof(TaskDefinitionManager))]
+    public interface ITaskDefinitionManager : IAgentService
+    {
+        Task DownloadAsync(IExecutionContext executionContext, IEnumerable<TaskInstance> tasks);
+
+        Definition Load(TaskReference task);
+    }
+
+    public sealed class TaskDefinitionManager : AgentService, ITaskDefinitionManager
+    {
+        private async Task DownloadAsync(IExecutionContext executionContext, TaskReference task)
```

Comment:
The order of these methods is messed up. I was trying to leave them as is, but now that the file has been renamed it's all green anyway.

I'll fix the order.